src/sif/pedestriancost.cc

```diff
@@ -151,6 +156,54 @@ BaseCostingOptionsConfig GetBaseCostOptsConfig() {
 
 const BaseCostingOptionsConfig kBaseCostOptsConfig = GetBaseCostOptsConfig();
 
+// Speed adjustment factors based on weighted grade. Comments here show an
+// example of speed changes based on "grade", using a base speed of 5 KPH
+// on flat roads. 
+// Tobler seems a bit too "fast" uphill so we use
+// but downhill Tobler seems good https://mtntactical.com/research/yet-calculating-movement-uneven-terrain/
+constexpr float kGradeBasedSpeedFactor[] = {
+    0.67f,  // -10%  - 4.71
+    0.71f,  // -8%   - 4.4
+    0.75f,  // -6.5% - 4.17
+    0.8f,  // -5%   - 3.96
+    0.85f,  // -3%   - 3.69
+    0.90f,  // -1.5% - 3.5
+    1.0f,  // 0%    - 3.16
+    1.06f, // 1.5%  - 3.15
+    1.12f, // 3%    - 2.99
+    1.21f, // 5%    - 2.79
+    1.30f, // 6.5%  - 2.65
+    1.37f, // 8%    - 2.51
+    1.49f,  // 10%   - 2.34
+    1.59f, // 11.5% - 2.22
+    1.69f,  // 13%   - 2.11
+    1.82f   // 15%   - 1.97
+};
+
+// Avoid hills "strength". How much do we want to avoid a hill. Combines
+// with the usehills factor (1.0 - usehills = avoidhills factor) to create
+// a weighting penalty per weighted grade factor. This indicates how strongly
+// edges with the specified grade are weighted. Note that speed also is
+// influenced by grade, so these weights help further avoid hills.
+constexpr float kAvoidHillsStrength[] = {
```

Comment:
i can see us wanting to change the shape of this curve depending on the mode of travel in the future like for wheelchair users for example. we can save that for another day though, just wanted to mention it